### PR TITLE
Feature/445 stretchfield

### LINKED ISSUE #445: [iOS] Timetable Grid: Double Field events
There is an even type "RoomIJ" whose usage and display type is unclear.

1) Find out if we actually need to implement this for timetable
2) If we need to implement it, do so

### PR DESCRIPTION
**Pull request by charles-b-stb**
## Issue
- close #445

## Overview (Required)
- Attempted to stretch cell across 5 columns in cases that appear to meet the situations shown in the figma design
- ~~TODO: The cell spacing needs adjustment.~~
- ~~TODO: The stretched cells should have the room color removed.~~ colors fixed

Based off https://github.com/DroidKaigi/conference-app-2024/pull/367 and should not be merged until after the grid merge is complete.

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :--:
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="" width="300" >

### CODE REVIEW

**Comment by charles-b-stb** [comment]
This still needs more work. For some reason other fields are getting double entries.

**Comment by charles-b-stb** [comment]
Fixed build issue (from merge) and fixed colors for extended fields. Ready for some prelim review... what do folks think of this kind of implementation?

**Comment by charles-b-stb** [comment]
After talking discussions about what the website is doing, I decided to try and extend just the lunch field. Still needs to be discussed with the mobile team to see if this change is worth keeping.

**Comment by charles-b-stb** [comment]
Let's go ahead and review this one and merge if possible.

File: app-ios/Sources/TimetableFeature/TimetableGridCard.swift [hunk 1]
```diff
@@ -7,13 +7,16 @@ import CommonComponents
 public struct TimetableGridCard: View {
     let timetableItem: TimetableItem
     let onTap: (TimetableItem) -> Void
+    let cellCount: Int
     
     public init(
         timetableItem: TimetableItem,
+        cellCount: Int? = 1,
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableGridCard.swift:14]: I think this argument is no needed an optional 🤔
and required property is no needed a default value 👍
```suggestion
        cellCount: Int,
```

File: app-ios/Sources/TimetableFeature/TimetableGridCard.swift [hunk 2]
```diff
@@ -22,17 +25,19 @@ public struct TimetableGridCard: View {
...
                 HStack(spacing: 4) {
-                    RoomTypeShape(roomType: .init(enTitle: timetableItem.room.name.enTitle))
+                    if cellCount == 1 {
+                        RoomTypeShape(roomType: .init(enTitle: timetableItem.room.name.enTitle))
                         .foregroundStyle(timetableItem.room.roomTheme.primaryColor)
+                    }
                     Text("\(timetableItem.startsTimeString) - \(timetableItem.endsTimeString)")
                         .textStyle(.labelMedium)
-                        .foregroundStyle(timetableItem.room.roomTheme.primaryColor)
+                        .foregroundStyle(cellCount>1 ? AssetColors.Surface.onSurfaceVariant.swiftUIColor : timetableItem.room.roomTheme.primaryColor)
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableGridCard.swift:34]:
```suggestion
                        .foregroundStyle(cellCount > 1 ? AssetColors.Surface.onSurfaceVariant.swiftUIColor : timetableItem.room.roomTheme.primaryColor)
```
I want add spaces for any operators left and right😅

File: app-ios/Sources/TimetableFeature/TimetableListView.swift [hunk 3]
```diff
@@ -198,7 +209,7 @@ fileprivate var bottomTabBarPadding: some View {
 }
 
 extension RoomType {
-    func toRoom() -> TimetableRoom {
+    public func toRoom() -> TimetableRoom {
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableListView.swift:212]: Why did you add this `public`🤔？
I seem this PR doesn't reference this method.

**Review by MrSmart00 — COMMENTED** [review]
Thanks!!
I left some comments!
Please check them🙏

File: app-ios/Sources/TimetableFeature/TimetableListView.swift (hunk 3 above)
Comment (charles-b-stb) [reply app-ios/Sources/TimetableFeature/TimetableListView.swift]: I was using it elsewhere, but on closer inspection it was in the same file so public was not needed.

**Comment by charles-b-stb** [comment]
@MrSmart00  Thanks for your comments! I responded and made some adjustments.

File: app-ios/Sources/TimetableFeature/TimetableListView.swift [hunk 4]
```diff
@@ -265,9 +276,9 @@ extension RoomType {
 }
 
 extension TimetableTimeGroupItems {
-    func getCellForRoom(room: RoomType, onTap: @escaping (TimetableItemWithFavorite) -> Void) -> TimetableGridCard? {
+    func getCellForRoom(room: RoomType, cellCount: Int?=1, onTap: @escaping (TimetableItemWithFavorite) -> Void) -> TimetableGridCard? {
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableListView.swift:279]:
```suggestion
    func getCellForRoom(room: RoomType, cellCount: 1, onTap: @escaping (TimetableItemWithFavorite) -> Void) -> TimetableGridCard? {
```

File: app-ios/Sources/TimetableFeature/TimetableGridCard.swift [hunk 5]
```diff
@@ -7,13 +7,16 @@ import CommonComponents
...
+    let cellCount: Int
     
     public init(
         timetableItem: TimetableItem,
+        cellCount: Int,
         onTap: @escaping (TimetableItem) -> Void
     ) {
         self.timetableItem = timetableItem
         self.onTap = onTap
+        self.cellCount = cellCount ?? 1
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableGridCard.swift:19]: you have `?? 1` is never used used warning yet.
you should remove it👍

**Review by MrSmart00 — COMMENTED** [review]
left some comments~
please check them🙏

File: app-ios/Sources/TimetableFeature/TimetableListView.swift (hunk 4 above)
Comment (charles-b-stb) [reply app-ios/Sources/TimetableFeature/TimetableListView.swift:279]: I might be wrong, but I think the intended change is removing "?=1" not the Int part... correct?

Comment (MrSmart00) [reply app-ios/Sources/TimetableFeature/TimetableListView.swift]: sorry it's wrong😅
```
func getCellForRoom(room: RoomType, cellCount: Int, onTap: @escaping (TimetableItemWithFavorite) -> Void) -> TimetableGridCard? {
```
is correct👍
@charles-b-stb

File: app-ios/Sources/TimetableFeature/TimetableGridCard.swift [hunk 6]
```diff
@@ -58,17 +63,17 @@ public struct TimetableGridCard: View {
             }
             .frame(maxWidth: .infinity)
             .padding(12)
-            .frame(width: 192, height: 153)
-            .background(timetableItem.room.roomTheme.containerColor, in: RoundedRectangle(cornerRadius: 4))
-            .overlay(RoundedRectangle(cornerRadius: 4).stroke(timetableItem.room.roomTheme.primaryColor, lineWidth: 1))
+            .frame(width: 192*CGFloat(cellCount)+CGFloat(12*(cellCount-1)), height: 153)
```
Comment (MrSmart00) [inline app-ios/Sources/TimetableFeature/TimetableGridCard.swift:66]:
```suggestion
            .frame(width: 192 * CGFloat(cellCount) + CGFloat(12 * (cellCount - 1)), height: 153)
```
